src/ensureArray.js

```diff
@@ -0,0 +1,23 @@
+import { when, of } from 'ramda';
+import { isNotArray } from './index';
+
+
+/**
+ * Returns an array, even if not, wrap it in a singleton one.
+ *
+ * @func ensureArray
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.6.0|v2.6.0}
+ * @category List
+ * @sig a | [a] -> [a]
```

Comment:
I think this signature should be `* -> [*]`, but not sure.

Reply:
`R.of` signature is `a → [a]`, so I did according to that.

Reply:
@char0n, need advice for this.

Reply:
The signature is correct. It's called [parametrized types](https://github.com/ramda/ramda/wiki/Type-Signatures#parameterized-types).

Reply:
TBH it was more the `a | [a]` that looked wrong to me. I could only find examples where it was used on output, not in input, but good to know it's OK. Sorry @guillaumearm

Reply:
@Undistraction check [assocPath](http://ramdajs.com/docs/#assocPath)

It's bassically the same if we don't extract the union type outside

```js
[Idx] → a → {a} → {a}
Idx = String | Int
```

```js
[String | Int] → {a} → {a}
```